src/main/java/dbot/storage/Storage.java

```diff
@@ -0,0 +1,148 @@
+package dbot.storage;
+
+import dbot.command.Command;
+import dbot.exception.DBotException;
+import dbot.parser.Parser;
+import dbot.tasklist.TaskList;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+/**
+ * The Storage class handles all file reading and writing operations that DBot might have.
+ */
+public class Storage {
+    /** Default file path used if the user does not provide a file name. */
+    public static final String DEFAULT_STORAGE_FILEPATH = "addressbook.txt";
+    public final Path path;
+
+    /**
+     * Initializes a default Storage instance that uses the default file path.
+     *
+     * @throws InvalidStorageFilePathException if the default path is invalid
+     */
+    public Storage() throws InvalidStorageFilePathException {
+        this(DEFAULT_STORAGE_FILEPATH);
+    }
+
+    /**
+     * Initializes a Storage instance for the given file path.
+     *
+     * @param filePath A String providing the location of the file to read and write data to.
+     * @throws InvalidStorageFilePathException If the given file path is invalid.
+     */
+    public Storage(String filePath) throws InvalidStorageFilePathException {
+        path = Paths.get(filePath);
+        if (!isValidPath(path)) {
+            throw new InvalidStorageFilePathException("Storage file should end with '.txt'");
+        }
+    }
+
+    /**
+     * Saves the given TaskList by writing it to a file, creating one if one does not already exist.
+     *
+     * @param tasks The TaskList to be saved.
+     * @throws StorageOperationException If the file cannot be written to.
+     */
+    public void save(TaskList tasks) throws StorageOperationException {

```

Comment:
Will saveTasks be better here?

Reply:
Yes, that would be a more expressive name for the method and ill amend it accordingly.